lib/internal/mime.js

```diff
@@ -182,103 +298,103 @@ class MIMEParams {
+      marker.position++;
+
+      const nextChar = StringPrototypeCharCodeAt(str, marker.position);
+
+      // #11-7 Let parameterValue be null.
       let parameterValue = null;
-      if (char === '"') {
-        // Handle quoted-string form of values
-        // skip '"'
-        position += 1;
-        // Find matching closing '"' or end of string
-        //   use $1 to see if we terminated on unmatched '\'
-        //   use $2 to see if we terminated on a matching '"'
-        //   so we can skip the last char in either case
-        const insideMatch = RegExpPrototypeExec(
-          QUOTED_VALUE_PATTERN,
-          StringPrototypeSlice(str, position));
-        position += insideMatch[0].length;
-        // Skip including last character if an unmatched '\' or '"' during
-        // unescape
-        const inside = insideMatch[1] || insideMatch[2] ?
-          StringPrototypeSlice(insideMatch[0], 0, -1) :
-          insideMatch[0];
+      // #11-8 If the code point at position within input is U+0022 ("), then:
+      if (nextChar === QUOTE_CHARCODE) {
+        // #11-8-1 If the code point at position within input is U+0022 ("), then:
         // Unescape '\' quoted characters
-        parameterValue = removeBackslashes(inside);
-        // If we did have an unmatched '\' add it back to the end
-        if (insideMatch[1]) parameterValue += '\\';
+        parameterValue = collectHTTPQuotedString(str, marker);
+
+        // #11-8-1 Collect a sequence of code points that are not U+003B (;) from input, given position.
+        const remaining = StringPrototypeIndexOf(str, SEMICOLON, marker.position);
+        marker.position += remaining !== -1 ? remaining - marker.position : 0;
       } else {
-        // Handle the normal parameter value form
-        const valueEnd = StringPrototypeIndexOf(str, SEMICOLON, position);
+        // #11-9-1 Set parameterValue to the result of
+        // collecting a sequence of code points that are not U+003B (;) from input,
+        // given position.
+        const valueEnd = StringPrototypeIndexOf(str, SEMICOLON, marker.position);
         const rawValue = valueEnd === -1 ?
-          StringPrototypeSlice(str, position) :
-          StringPrototypeSlice(str, position, valueEnd);
-        position += rawValue.length;
-        const trimmedValue = StringPrototypeSlice(
-          rawValue,
-          0,
-          SafeStringPrototypeSearch(rawValue, START_ENDING_WHITESPACE),
-        );
-        // Ignore parameters without values
-        if (trimmedValue === '') continue;
+          StringPrototypeSlice(str, marker.position) :
+          StringPrototypeSlice(str, marker.position, valueEnd);
+
+        // #11-9-2 Remove any trailing HTTP whitespace from parameterValue.
+        const trimmedValue = StringPrototypeTrimEnd(rawValue);
+        marker.position += trimmedValue.length;
+
+        // #11-9-3 If parameterValue is the empty string, then continue.
+        if (trimmedValue.length === 0) continue;
```

Comment:
switch positions as if trimmedValue.length === 0 means that marker.position += trimmedValue.length is doing nothing.

```suggestion
        // #11-9-3 If parameterValue is the empty string, then continue.
        if (trimmedValue.length === 0) continue;

                marker.position += trimmedValue.length;

```